include/etc.hpp

```diff
@@ -38,4 +38,105 @@ std::string string_format( const std::string& format, Args ... args )
     return std::string( buf.get(), buf.get() + size - 1 ); // We don't want the '\0' inside
 }
 
+class TimeProfiler
+{
+private:
+    Array<StopWatch *> timers;
+
+    Array<int> calls;
+    Array<bool> starts;
+    std::vector<std::string> names;
+    std::unordered_map<std::string, int> indices;
+
+    const MPI_Comm comm;
+    int rank;
+
+public:
+    TimeProfiler(MPI_Comm comm_=MPI_COMM_WORLD)
+        : comm(comm_), timers(0), starts(0)
```

Comment:
`Array` has size to `0` by default constructor, so there is no need to initialize them this way.